File: python/spec/support_spec/file_to_list_spec.py
#!/usr/bin/env python3
#
from mamba import description, context, it
from expects import expect, equal

import spec.support

with description('file_to_list') as self:
  with before.all:
    self.subject = spec.support.file_to_list(__file__.replace('_spec.py', '.txt'))

  with it("three lines"):
    expect(len(self.subject)).to(equal(3))

  with it("separates lines"):
    expect(self.subject).to(equal(["payter", "is a", "hosehead"]))

File: c/spec/support_spec/file_to_list_spec.c
#include <aoc2025.h>

#include "file_to_list.h"

#include "bdd-for-c.h"

spec("file_to_list") {
  static list_t *subject = NULL;

  describe("dunsels") {
    before() {
      subject = file_to_list("spec/support_spec/file_to_list.txt");
    }

    it("three lines") {
      check(subject->next->next->next == NULL);
    }

    it("separates lines") {
      check(!(strcmp("payter", subject->text)));
      check(!(strcmp("is a", subject->next->text)));
      check(!(strcmp("hosehead", subject->next->next->text)));
    }
  }
}

